Dockerfile

```diff
@@ -1,36 +1,21 @@
-# take default image of node boron i.e  node 6.x
-FROM node:8.9 as builder
-RUN npm i -g yarn
-
-# create app directory in container
-RUN mkdir -p /app
+# node image
+FROM node:8.14.0-alpine
 
 # set /app directory as default working directory
-WORKDIR /app
-
-# only copy package.json initially so that `RUN yarn` layer is recreated only
-# if there are changes in package.json
+WORKDIR /app/
 COPY . /app/
 
-RUN yarn
-
-# compile to ES5
-RUN yarn build
-
-FROM node:8.9 
-
-WORKDIR /app
-
-COPY --from=builder /app /app
+# Obtain ssh-keygen and patch for snyk
+RUN apk update && apk add --no-cache openssh-keygen && apk add --no-cache openssl && apk add --no-cache patch
 
 # set up public and private keys
-RUN echo -e 'y\n'|ssh-keygen -q -t rsa -b 4096 -N "" -f private.key &&\
+RUN echo -e 'y\n'|ssh-keygen -q -t rsa -b 4096 -N "" -f private.key && \
     openssl rsa -in private.key -pubout -outform PEM -out private.key.pub
 
-#USER 50000:50000
+RUN yarn install --pure-lockfile && yarn build && yarn install --production --frozen-lockfile
```

Comment:
Using the [multi-stage build](https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds) means that we no longer need to include the install/build/install on the same line.